components/Dialogs/src/private/MDCAlertControllerView+Private.m

```diff
@@ -547,26 +579,46 @@ - (CGSize)calculateContentSizeThatFitsWidth:(CGFloat)boundingWidth {
   return contentSize;
 }
 
-// @param boundingWidth should not include any internal margins or padding
+/**
+ Calculate the size of the title frame, which includes an optional title, optional title icon and
+ optional icon view.
+
+ @param boundingWidth should not include any internal margins or padding
+*/
 - (CGSize)calculateTitleViewSizeThatFitsWidth:(CGFloat)boundingWidth {
+  CGFloat leftInset =
+      self.enableAdjustableInsets
+          ? MAX(MAX(self.titleInsets.left, self.titleIconInsets.left), self.contentInsets.left)
+          : MDCDialogContentInsets.left;
+  CGFloat rightInset =
+      self.enableAdjustableInsets
+          ? MAX(MAX(self.titleInsets.right, self.titleIconInsets.right), self.contentInsets.right)
+          : MDCDialogContentInsets.right;
+
   CGSize boundsSize = CGRectInfinite.size;
-  boundsSize.width = boundingWidth - MDCDialogContentInsets.left - MDCDialogContentInsets.right;
+  boundsSize.width = boundingWidth - leftInset - rightInset;
 
   CGSize titleSize = [self.titleLabel sizeThatFits:boundsSize];
   CGSize messageSize = [self.messageLabel sizeThatFits:boundsSize];
   CGSize accessoryViewSize = [self.accessoryView systemLayoutSizeFittingSize:boundsSize];
   CGFloat contentWidth = MAX(MAX(titleSize.width, messageSize.width), accessoryViewSize.width) +
-                         MDCDialogContentInsets.left + MDCDialogContentInsets.right;
+                         leftInset + rightInset;
 
-  CGFloat contentHeight = MDCDialogContentInsets.top + [self titleIconViewSize].height +
-                          [self contentTitleIconVerticalPadding] + titleSize.height +
-                          [self contentInternalVerticalPadding];
+  CGFloat totalElementsHeight = [self titleIconViewSize].height + titleSize.height;
 
-  CGSize contentSize;
-  contentSize.width = (CGFloat)ceil(contentWidth);
-  contentSize.height = (CGFloat)ceil(contentHeight);
+  // Adjustable insets layout
+  CGFloat titleTotalHeight = totalElementsHeight + [self titleInsetTop] +
+                             [self titleIconInsetBottom] + [self titleInsetBottom];
+  CGSize titleTotalSize = CGSizeMake((CGFloat)ceil(contentWidth), (CGFloat)ceil(titleTotalHeight));
 
-  return contentSize;
+  // Fixed insets layout
+  CGFloat fixedInsetsHeight = totalElementsHeight + MDCDialogContentInsets.top +
+                              [self fixedInsetsContentTitleIconVerticalPadding] +
+                              [self contentInternalVerticalPadding];
+  CGSize fixedInsetsTitleSize =
+      CGSizeMake((CGFloat)ceil(contentWidth), (CGFloat)ceil(fixedInsetsHeight));
```

Comment:
What is the reason for using ceil here? Can we avoid it?

Reply:
If we take if out, we may have some differences in snapshot tests. Not sure what the original reason was. I think they wanted the different sections of the dialogs to be able to conform to specs, which are typically whole numbers.